File: src/test/java/tool/cmnclslib/cls/UnitTest_ClsAdler32.java
package tool.cmnclslib.cls;

import java.io.ByteArrayInputStream;
import java.nio.charset.StandardCharsets;
import org.junit.Test;
import static org.junit.Assert.*;

/**
 * ClsAdler32 の単体テストクラスです。
 */
public class UnitTest_ClsAdler32 {

    public static final String SAMPLE_DOCUMENT = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
    public static final String ADLER_VALUE = "376443647";

    private ClsAdler32 objAdler32 = new ClsAdler32();

    @Test
    public void testAdler32Calculation() {
        assertEquals(ADLER_VALUE, objAdler32.computeChecksum(new ByteArrayInputStream(SAMPLE_DOCUMENT.getBytes(StandardCharsets.US_ASCII))));
    }
}

File: TestProject1/Class/UnitTest_ClsAdler32.cs
using CmnClsLib.Module;
using CmnClsLib.Class;
using CmnClsLib.Interface;
using System.Text;
using Xunit;
using Assert = Xunit.Assert;

namespace TestProject1.Class
{
    public class UnitTest_ClsAdler32
    {
        public const String SampleDocument = @"abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
        public const String AdlerValue = "376443647";

        private ClsAdler32 _objAdler32 = new();

        // --------------------------------------------------------------------
        // public String ComputeChecksum(String strPathFFile)
        // --------------------------------------------------------------------
        [Fact]
        public void abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZのadler32値が376443647であること()
        {
            // 第1引数：期待する値
            // 第2引数：計算結果
            Assert.Equal(AdlerValue, _objAdler32.ComputeChecksum(new MemoryStream(Encoding.ASCII.GetBytes(SampleDocument))));
        }
    }
}
